third_party/cached_httpfs/http_file_cache.cpp

```diff
@@ -48,6 +48,17 @@ CachedFileHandle::CachedFileHandle(shared_ptr<CachedFile> &file_p) {
 	file = file_p;
 }
 
+void CachedFileHandle::WriteMetadata(const string &cache_key, const string &remote_path, idx_t total_size) {
+	D_ASSERT(!file->initialized);
+	string metadata_file_name = file->cache_directory + "/" + cache_key + ".meta";
+	FileOpenFlags flags = FileFlags::FILE_FLAGS_WRITE | FileFlags::FILE_FLAGS_FILE_CREATE | FileLockType::WRITE_LOCK;
+	auto handle = file->fs.OpenFile(metadata_file_name, flags);
+	string metadata_info = cache_key + "," + remote_path + "," + std::to_string(total_size);
+	handle->Write((void *)metadata_info.c_str(), metadata_info.length(), 0);
```

Comment:
Doesn't `Write` take a `const char*`?
```suggestion
	handle->Write(cache_key.c_str(), cache_key.length(), 0);
	handle->Write(",", 1, 0);
	handle->Write(remote_path.c_str(), remote_path.length(), 0);
	handle->Write(",", 1, 0);
	auto s = std::to_string(total_size);
	handle->Write(s.c_str(), s.length(), 0);
```

I'm also surprised we don't need a new line after?

Reply:
Write complains unless it is void * so we need to cast. Also, writing commas will also fail.
IMO, constructing single metadata line is simpler.

Reply:
Ok, fair enough